lib/src/rules/prefer_constructors_over_static_methods.dart

```diff
@@ -78,10 +78,12 @@ class _Visitor extends SimpleAstVisitor<void> {
   void visitMethodDeclaration(MethodDeclaration node) {
     final returnType = node.returnType?.type;
     final parent = node.parent;
+    final typeParameters = node.typeParameters?.typeParameters ?? [];
     if (node.isStatic &&
         parent is ClassOrMixinDeclaration &&
         returnType is InterfaceType &&
-        returnType.element == parent.declaredElement &&
+        returnType == parent.declaredElement.thisType &&
```

Comment:
If I'm understanding this correctly, this line will be true only when `parent` has no type parameters (isn't generic), so this effectively disables the lint for generic classes. I have to wonder, though whether it might be more clear to future readers to simply ask whether the class declaration has type parameters (`parent.typeParameters != null`).

Reply:
Done